lua/entities/cfc_rotten_tomato_projectile/init.lua

```diff
@@ -100,13 +89,25 @@ function ENT:Touch( ent )
 
     local speed = self:GetVelocity():Length()
 
-    self:HitEffects( pos, normal, speed )
+    local caughtDamage
+    local damageDealt = 0
+    local hookName = "cfc_throwable_projectile_damagecatch_" .. self:GetCreationID()
+
+    -- do anything that needs to respect hooks, in PostHitEnt
+    hook.Add( "PostEntityTakeDamage", hookName, function( victim, dmgInfo )
+        local inflictor = dmgInfo:GetInflictor()
+        if inflictor ~= self then return end
+        hook.Remove( "PostEntityTakeDamage", hookName ) -- remove the hook so only gets called once
+        caughtDamage = true
+        damageDealt = dmgInfo:GetDamage()
+        self:PostHitEnt( victim, damageDealt )
+
+    end )
```

Comment:
```suggestion
    local damageDealt = 0
    local actuallyDidDamage = false
    local hookName = "cfc_throwable_projectile_damagecatch_" .. self:GetCreationID()

    -- do anything that needs to respect hooks, in PostHitEnt
    hook.Add( "PostEntityTakeDamage", hookName, function( victim, dmgInfo, took )
        local inflictor = dmgInfo:GetInflictor()
        if inflictor ~= self then return end
        hook.Remove( "PostEntityTakeDamage", hookName ) -- remove the hook so only gets called once
        caughtDamage = true
        damageDealt = dmgInfo:GetDamage()
        actuallyDidDamage = took
        self:PostHitEnt( victim, damageDealt, actuallyDidDamage )
    end )
```

And likewise update the `:PostHit()` call below, and update the `PostHit` and `PostHitEnt` functions to use `actuallyDidDamage` accordingly.